code/go/0chain.net/smartcontract/zcnsc/mint.go

```diff
@@ -114,11 +113,43 @@ func (zcn *ZCNSmartContract) Mint(trans *transaction.Transaction, inputData []by
 	// record the global nonce from solidity smart contract
 	gn.WZCNNonceMinted[payload.Nonce] = true
 
+	var (
+		amount currency.Coin
+		n      currency.Coin
+		share  currency.Coin
+	)
+	share, _, err = currency.DistributeCoin(gn.ZCNSConfig.MaxFee, int64(len(payload.Signatures)))
+	if err != nil {
+		err = errors.Wrap(err, fmt.Sprintf("%s, DistributeCoin operation, %s", code, info))
+		return
+	}
+	n, err = currency.Int64ToCoin(int64(len(payload.Signatures)))
```

Comment:
Unnecessary converting as we don't need the `n`

Reply:
we can calculate product in uint64 and then convert it to coin, will it be any different?